### PR TITLE
Add graph parameter to tf.import_graph_def()

### PR DESCRIPTION
**Pull request by samjabrahams**
Allows user to indicate a specific `Graph` to place `GraphDef` Operations. This keeps users from needing to use a `with g.as_default()` statement in their code. The documentation for `import_graph_def()` has also been improved to inform users that the default graph is used.. by default. For tests, I've simply done an altered version of the basic test thus far- let me know what further testing you'd like to see.

This should maintain backwards compatibility.

### CODE REVIEW

File: tensorflow/python/framework/importer.py [hunk 1]
```diff
@@ -232,232 +237,237 @@ def import_graph_def(graph_def, input_map=None, return_elements=None,
...
-
-          if input_name in input_map:
-            # (b) Input should be replaced by a tensor from the caller.
-            source_tensor = input_map[input_name]
-            used_input_keys.add(input_name)
-
-          else:
-            # (c) Input should be taken from an op in `graph_def`.
-            operation_name, output_index = _ParseTensorName(input_name)
+  g = ops.get_default_graph() if graph is None else graph
```
Comment (samjabrahams) [inline tensorflow/python/framework/importer.py]: Actual new piece of code here, pretty much everything below this line (aside from `with g.as_default():`) is a diff caused by indentation and tweaks to keep the column width below 80.

File: tensorflow/python/framework/importer.py [hunk 2]
```diff
@@ -232,232 +237,237 @@ def import_graph_def(graph_def, input_map=None, return_elements=None,
...
+  with g.as_default():
+    with ops.name_scope(name, 'import', input_map.values()) as scope:
```
Comment (samjabrahams) [inline tensorflow/python/framework/importer.py]: BEGIN THE WHITE SPACE DIFF

File: tensorflow/python/framework/importer.py [hunk 3]
```diff
@@ -232,232 +237,237 @@ def import_graph_def(graph_def, input_map=None, return_elements=None,
...
+              raise ValueError(
+                  'Requested return_element %r not found in graph_def.' % name)
+          else:
+            try:
+              ret.append(name_to_op[name])
+            except KeyError:
+              raise ValueError(
+                  'Requested return_element %r not found in graph_def.' % name)
+        return ret
+    # LINT.ThenChange(//tensorflow/core/graph/graph_constructor.cc)
```
Comment (samjabrahams) [inline tensorflow/python/framework/importer.py]: END THE WHITE SPACE DIFF

File: tensorflow/python/framework/importer_test.py [hunk 4]
```diff
@@ -219,6 +219,72 @@ def testBasic(self):
       # Check that the op_def is still available.
       self.assertNotEqual(None, a.op_def)
 
+      # Check that the Ops were placed in the graph
```
Comment (samjabrahams) [inline tensorflow/python/framework/importer_test.py]: Added some tests to `testBasic` to make sure the added Ops were in the default graph.

File: tensorflow/python/framework/importer_test.py [hunk 5]
```diff
@@ -219,6 +219,72 @@ def testBasic(self):
...
+      self.assertEqual(a, g.get_operation_by_name("import/A"))
+      self.assertEqual(b, g.get_operation_by_name("import/B"))
+      self.assertEqual(c, g.get_operation_by_name("import/C"))
+      self.assertEqual(d, g.get_operation_by_name("import/D"))
+
+  def testBasicGraphParam(self):
```
Comment (samjabrahams) [inline tensorflow/python/framework/importer_test.py]: Basically a copy/paste of `testBasic`, without a call to `as_default()`

**Comment by martinwicke** [comment]
Ok, we looked at this -- since we don't take graph arguments on most functions, and since this is the same behavior as 

```
with g.as_default():
  tf.import_graph_def(...)
```

We probably don't want to add the extra argument here.

**Comment by samjabrahams** [comment]
@martinwicke what are your thoughts on tweaking the documentation to indicate that the imported Ops get placed into the default graph? Something like making the first line read "Imports the TensorFlow graph in `graph_def` into the _current default_ Python `Graph`".

**Comment by martinwicke** [comment]
I think that would be a good idea.

**Comment by samjabrahams** [comment]
Submitted the documentation change as a PR in #7325

**Pull request closed without merging**